src/modules/billing/services/batch-service.js

```diff
@@ -268,6 +269,47 @@ const createChargeModuleBillRun = async batchId => {
   return batch.pickFrom(row, ['externalId', 'billRunId']);
 };
 
+const getTransactionsWithTwoPartError = batch => {
+  return batch.invoices.reduce((acc, invoice) => {
+    invoice.invoiceLicences.forEach(invoiceLicence => {
+      invoiceLicence.transactions.forEach(transaction => {
+        if (transaction.twoPartTariffError) acc.push(transaction);
+        return acc;
+      });
+    });
+    return acc;
+  }, []);
+};
+
```

Comment:
I think I'd find this a bit easier to understand if we split the logic out into some extra functions:
```
const assertNoTransactionsWithTwoPartError = batch => {
  const transactions = getTransactionsWithTwoPartError(batch);
  if(transactions.length) {
    throw new TransactionStatusError('Cannot approve review. There are outstanding two part tariff errors to resolve');
   }
}

const assertBatchStatusIsReview = batch => {
  if(batch.status !== BATCH_STATUS.review) {
    throw new BatchStatusError('Cannot approve review. Batch status must be "review"');
  } 
}
```

Then the function below can do:
```
const approveTptBatchReview = async batch => {
  assertNoTransactionsWithTwoPartError(batch);
  assertBatchStatusIsReview(batch);
  await setStatus(batch.id, BATCH_STATUS.processing);
  return getBatchById(batch.id);
}
```